### ISSUE TITLE
Number of SNPs for each threshold

### ISSUE THREAD

**Issue by Anbupalam** [choishingwan/PRSice#40]
I am using the latest version of PRSice. 

By default, the score for the p-value threshold 1.0 is produced including the score for the thresholds set in the --bar-levels parameter.

Also the number of SNPs used for each threshold calculation is not available even if I set the --print-snp option.

With the same set of parameters used in the previous version and the current version, the score are not exactly the same, although the correlation is very high (0.986). Is there a reason for this change?

Thanks
Anbu.

**Comment by choishingwan** [comment]
Hi,

The number of SNP used should be included in the *.prsice file whereas the —print-snp option will generate a *.snps file containing all the SNPs after clumping and a column indicating if they are included in the best threshold

P value of 1 is now included as a default. You can disable this by using —no-default

There is a difference in the PRS because we are now doing all the calculation of test statistics within the C++ program as opposed to using awk scripts. This allow us to have higher precision, which as large number of SNP culminated, this can add up to difference in PRS though the difference should be smaller. Also, previous version have occational problem in flipping where a very small amount of SNPs can be flipped incorrectly, which can also leads to difference

**Comment by Anbupalam** [comment]
Thanks for coming back quickly on this.

I am using the following options in my script:
--A1 A1 \
--A2 A2 \
--bp BP \
--chr CHR \
--pvalue P \
--snp SNP \
--beta T \
--stat BETA \
--se SE \
--prsice /data2/anbupalam/PRSice2/PRSice_linux \
--base mydatafile.txt \
--target myinputfile \
--thread 1 \
--bar-levels 0.5 \
--clump-kb 250 \
--clump-r2 0.25 \
--clump-p 1.0 \
--all \
--print-snp \
--binary-target F \
--no-regress \
--fastscore \
--score no_mean_imputation \

I don't get any *.snps output file mentioned in your reply. 
Could you please clarify regarding the number of SNPs to be included in the *.prsice file?

Thanks
Anbu.

**Comment by choishingwan** [comment]
Judging on your script, you are not using the latest version. Maybe you can download the latest version and re run PRSice again. For more information about the output, you can read our documentation [here](http://prsice.readthedocs.io/en/latest/)

(As you didn't use the **--out** option, the SNP file should be named **PRSice.snps**, and
For number of SNP included in each threshold, the **PRSice.prsice** file should contain a column call *Num_SNP* which provide the number of SNPs included in each P-value threshold

**Comment by Anbupalam** [comment]
I did download the recent version and re-run the code. I still don't get the PRSice.snps file. There is a summary file PRSice.summary which has these columns below but without any data. Probably the program produces these output only if the regression is done. Please check.

Phenotype       Set     Threshold       PRS.R2  Full.R2 Null.R2 Prevalence      Coefficient     Standard.Error  P       Num_SNP 

Thanks
Anbu.

**Comment by Anbupalam** [comment]
I am using the version 2.1.0 beta.

**Comment by choishingwan** [comment]
Then the program didn’t completed

Can you show us the log file?

**Comment by Anbupalam** [comment]
PRSice 2.1.0.beta (14 Jan 2018) 
https://github.com/choishingwan/PRSice
(C) 2016-2017 Shing Wan (Sam) Choi, Jack Euesden, Cathryn M. Lewis, Paul F. O'Reilly
GNU General Public License v3

If you use PRSice in any published work, please cite:
Jack Euesden Cathryn M. Lewis Paul F. O'Reilly (2015)
PRSice: Polygenic Risk Score software.
Bioinformatics 31 (9): 1466-1468

2018-01-31 17:44:52
/data/anbu/PRSice2/PRSice_linux \
    --A1 A1 \
    --A2 A2 \
    --all-score  \
    --bar-levels 0.500000,1.000000 \
    --base myfile.txt \
    --beta  \
    --binary-target F \
    --bp BP \
    --chr CHR \
    --clump-kb 250 \
    --clump-p 1.000000 \
    --clump-r2 0.250000 \
    --fastscore  \
    --model add \
    --no-default  \
    --no-regress  \
    --out Test \
    --print-snp  \
    --pvalue P \
    --se SE \
    --seed 1476414445 \
    --snp SNP \
    --stat BETA \
    --target myinput \
    --thread 1


Loading Genotype file: 
myinput
(bed) 
 

539 people (0 male(s), 0 female(s)) observed 
539 founder(s) included 
 

1874650 ambiguous variant(s) excluded 
11599982 variant(s) included 
 

1 region included 

Start processing myfile.txt 
============================== 
 

Base file: 
myfile.txt
11389 variant(s) observed in base file, with: 
1 ambiguous variant(s) excluded 
1532 variant(s) not found in target file 
1 mismatched variant(s) excluded 
9856 total variant(s) included from base file 
 
 

Start performing clumping 

24028 MB RAM detected; reserving 12014 MB for clumping 
 

Number of variant(s) after clumping : 2155 
 

There are a total of 1 phenotype to process 
 

539 sample(s) with valid phenotype 
 

There are  Please note that these results are inflated due 
to the overfitting inherent in finding the best-fit PRS 
(but it's still best to find the best-fit PRS!). 
You can use the --perm option (see manual) to calculate an 
empirical P-value.

**Comment by choishingwan** [comment]
Ar, now I see it. Sorry, should have spotted that earlier. When **--no-regress** is used. The only output from PRSice will be the all.score file, which didn't provide the number of SNPs for each threshold. 

I have now added the print-snp function for no-regress though it will only should you the SNPs post clumping and you will still need to calculate the number of SNPs per threshold based on that. The main reason behind is that the prsice output doesn't make sense when **--no-regress** is used (everything will be - ). 

Unfortunately, as our server is down, I cannot compile the executable until the server is back on. If you can't wait, you can try checkout the commit  b3323c4 and compile it yourself.

On linux computer, with g++ that supports c++11, you can compile PRSice using

``` bash
git clone https://github.com/choishingwan/PRSice.git
cd PRSice
git checkout b3323c4
g++ -std=c++11 -isystem lib -I inc src/*.cpp -lpthread -lz -o PRSice
```

(You will need zlib on your computer but it should usually be installed)

**Comment by Anbupalam** [comment]
Thanks you very much. It would be easy for me to work on the compiled binary file. Please let me know if you have compiled version ready.

**Comment by choishingwan** [comment]
Here you go:

https://github.com/choishingwan/PRSice/releases/download/2.1.0.beta/PRSice_linux_20180201.zip

(Note: The --print-snp will only give you the SNPs post clumping and you will still need to manually count the SNPs in each threshold, please remember that the thresholds are inclusive, e.g. threshold of 0.5 will include SNPs with p-value exactly at 0.5)

**Comment by JoniColeman** [comment]
I think this should be re-opened and adding number of SNPs at each threshold for --no-regress considered as a future enhancement.

**Comment by choishingwan** [comment]
Done, implemented in be2e0c3e3f77b0a50302c1058f19eb47fa2c8b6a, should be available in 2.2.6
